tests/utils.rs

```diff
@@ -35,4 +36,28 @@ mod utils {
 
         (Arc::new(sconfig), Arc::new(cconfig))
     }
+
+    #[allow(dead_code)]
+    pub async fn write<W: AsyncWrite + Unpin>(
+        w: &mut W,
+        data: &[u8],
+        vectored: bool,
+    ) -> io::Result<()> {
+        if vectored {
```

Comment:
Nit: would do early return here for the non-`vectored` path to reduce rightward drift.